Create clean Command Line applications

Because this is not very nice to say:

```
$ ./router
panic: strconv.ParseUint: parsing "": invalid syntax

goroutine 1 [running]:
main.parseOptions(0x0, 0x0, 0x0, 0x0, 0x32a080)
	/.../github.com/TheThingsNetwork/ttn/integration/router/main.go:105 +0x1e7
main.main()
	/.../github.com/TheThingsNetwork/ttn/integration/router/main.go:24 +0x27
```

--- a/integration/broker/main.go
+++ b/integration/broker/main.go
@@ -5,7 +5,6 @@ package main
 
 import (
 	"flag"
-	"fmt"
 	"os"
 	"strconv"
 
@@ -18,9 +17,6 @@ import (
 )
 
 func main() {
-	// Parse options
-	routersPort, handlersPort := parseOptions()
-
 	// Create Logging Context
 	log.SetHandler(text.New(os.Stdout))
 	log.SetLevel(log.DebugLevel)
@@ -28,25 +24,28 @@ func main() {
 		"component": "Broker",
 	})
 
+	// Parse options
+	routersPort, handlersPort := parseOptions()
+
 	// Instantiate all components
 	rtrAdapter, err := http.NewAdapter(uint(routersPort), http.JSONPacketParser{}, ctx.WithField("tag", "Routers Adapter"))
 	if err != nil {
-		panic(err)
+		ctx.WithError(err).Fatal("Could not start Routers Adapter")
 	}
 
 	hdlHTTPAdapter, err := http.NewAdapter(uint(handlersPort), http.JSONPacketParser{}, ctx.WithField("tag", "Handlers Adapter"))
 	if err != nil {
-		panic(err)
+		ctx.WithError(err).Fatal("Could not start Handlers Adapter")
 	}
 
 	hdlAdapter, err := pubsub.NewAdapter(hdlHTTPAdapter, pubsub.HandlerParser{}, ctx.WithField("tag", "Handlers Adapter"))
 	if err != nil {
-		panic(err)
+		ctx.WithError(err).Fatal("Could not start Handlers Adapter")
 	}
 
 	broker, err := components.NewBroker(ctx.WithField("tag", "Broker"))
 	if err != nil {
-		panic(err)
+		ctx.WithError(err).Fatal("Could not start Broker")
 	}
 
 	// Bring the service to life
@@ -56,12 +55,12 @@ func main() {
 		for {
 			packet, an, err := rtrAdapter.Next()
 			if err != nil {
-				fmt.Println(err)
+				ctx.WithError(err).Error("Could not retrieve uplink")
 				continue
 			}
 			go func(packet Packet, an AckNacker) {
 				if err := broker.HandleUp(packet, an, hdlAdapter); err != nil {
-					fmt.Println(err)
+					ctx.WithError(err).Error("Could not process uplink")
 				}
 			}(packet, an)
 		}
@@ -72,12 +71,12 @@ func main() {
 		for {
 			reg, an, err := hdlAdapter.NextRegistration()
 			if err != nil {
-				fmt.Println(err)
+				ctx.WithError(err).Error("Could not retrieve registration")
 				continue
 			}
 			go func(reg Registration, an AckNacker) {
 				if err := broker.Register(reg, an); err != nil {
-					fmt.Println(err)
+					ctx.WithError(err).Error("Could not process registration")
 				}
 			}(reg, an)
 		}
@@ -89,19 +88,30 @@ func main() {
 func parseOptions() (routersPort uint64, handlersPort uint64) {
 	var routersPortFlag string
 	var handlersPortFlag string
-	flag.StringVar(&routersPortFlag, "routers-port", "", "TCP port on which the broker should listen to for incoming uplink packets.")
-	flag.StringVar(&handlersPortFlag, "handlers-port", "", "TCP port on which the broker should listen to for incoming registrations and downlink packet.")
-	flag.Parse()
+
+	flags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	flags.StringVar(&routersPortFlag, "routers-port", "", "TCP port on which the broker should listen to for incoming uplink packets.")
+	flags.StringVar(&handlersPortFlag, "handlers-port", "", "TCP port on which the broker should listen to for incoming registrations and downlink packet.")
+
+	flags.Parse(os.Args[1:])
 
 	var err error
+
+	if routersPortFlag == "" {
+		log.Fatal("No Router listen port supplied using the -routers-port flag")
+	}
 	routersPort, err = strconv.ParseUint(routersPortFlag, 10, 64)
 	if err != nil {
-		panic(err)
+		log.Fatal("Could not parse the value for -routers-port")
 	}
 
+	if handlersPortFlag == "" {
+		log.Fatal("No Handler listen port supplied using the -handlers-port flag")
+	}
 	handlersPort, err = strconv.ParseUint(handlersPortFlag, 10, 64)
 	if err != nil {
-		panic(err)
+		log.Fatal("Could not parse the value for -handlers-port")
 	}
 
 	return

--- a/integration/router/main.go
+++ b/integration/router/main.go
@@ -20,9 +20,6 @@ import (
 )
 
 func main() {
-	// Parse options
-	brokers, tcpPort, udpPort := parseOptions()
-
 	// Create Logging Context
 	log.SetHandler(text.New(os.Stdout))
 	log.SetLevel(log.DebugLevel)
@@ -30,25 +27,28 @@ func main() {
 		"component": "Router",
 	})
 
+	// Parse options
+	brokers, tcpPort, udpPort := parseOptions()
+
 	// Instantiate all components
 	gtwAdapter, err := semtech.NewAdapter(uint(udpPort), ctx.WithField("tag", "Gateway Adapter"))
 	if err != nil {
-		panic(err)
+		ctx.WithError(err).Fatal("Could not start Gateway Adapter")
 	}
 
 	pktAdapter, err := http.NewAdapter(uint(tcpPort), http.JSONPacketParser{}, ctx.WithField("tag", "Broker Adapter"))
 	if err != nil {
-		panic(err)
+		ctx.WithError(err).Fatal("Could not start Broker Adapter")
 	}
 
 	brkAdapter, err := broadcast.NewAdapter(pktAdapter, brokers, ctx.WithField("tag", "Broker Adapter"))
 	if err != nil {
-		panic(err)
+		ctx.WithError(err).Fatal("Could not start Broker Adapter")
 	}
 
 	router, err := components.NewRouter(ctx.WithField("tag", "Router"))
 	if err != nil {
-		panic(err)
+		ctx.WithError(err).Fatal("Could not start Router")
 	}
 
 	// Bring the service to life
@@ -92,26 +92,37 @@ func parseOptions() (brokers []Recipient, tcpPort uint64, udpPort uint64) {
 	var brokersFlag string
 	var udpPortFlag string
 	var tcpPortFlag string
-	flag.StringVar(&brokersFlag, "brokers", "", `Broker addresses to which broadcast packets.
- 	For instance: 10.10.3.34:8080,thethingsnetwork.broker.com:3000
- 	`)
-	flag.StringVar(&udpPortFlag, "udp-port", "", "UDP port on which the router should listen to.")
-	flag.StringVar(&tcpPortFlag, "tcp-port", "", "TCP port on which the router should listen to.")
-	flag.Parse()
+
+	flags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	flags.StringVar(&brokersFlag, "brokers", "", `Broker addresses to which broadcast packets.
+ 	For instance: 10.10.3.34:8080,thethingsnetwork.broker.com:3000`)
+	flags.StringVar(&udpPortFlag, "udp-port", "", "UDP port on which the router should listen to.")
+	flags.StringVar(&tcpPortFlag, "tcp-port", "", "TCP port on which the router should listen to.")
+
+	flags.Parse(os.Args[1:])
 
 	var err error
+
+	if tcpPortFlag == "" {
+		log.Fatal("No TCP listen port supplied using the -tcp-port flag")
+	}
 	tcpPort, err = strconv.ParseUint(tcpPortFlag, 10, 64)
 	if err != nil {
-		panic(err)
+		log.Fatal("Could not parse the value for -tcp-port")
+	}
+
+	if udpPortFlag == "" {
+		log.Fatal("No UDP listen port supplied using the -udp-port flag.")
 	}
 	udpPort, err = strconv.ParseUint(udpPortFlag, 10, 64)
 	if err != nil {
-		panic(err)
-	}
-	if brokersFlag == "" {
-		panic("Need to provide at least one broker address")
+		log.Fatal("Could not parse the value for -udp-port")
 	}
 
+	if brokersFlag == "" {
+		log.Fatal("No broker address is supplied using -brokers flag.")
+	}
 	brokersStr := strings.Split(brokersFlag, ",")
 	for i := range brokersStr {
 		brokers = append(brokers, Recipient{

